src/fmu/dataio/_objectdata_provider.py

```diff
@@ -469,6 +481,52 @@ def _derive_spec_bbox_arrowtable(self):
 
         return specs, bbox
 
+    def _get_columns(self):
+        """Get the columns from table"""
+        if isinstance(self.obj, pd.DataFrame):
+            logger.debug("pandas")
+            columns = list(self.obj.columns)
+        else:
+            logger.debug("arrow")
+            columns = self.obj.column_names
+        logger.debug("Available columns in table %s ", columns)
+        return columns
+
+    def _derive_index(self):
+        """Derive table index"""
+        # This could in the future also return context
+        columns = self._get_columns()
+        index = []
+        preset_index = self.dataio.table_index
+        if preset_index is None:
+            logger.debug("Finding index to include")
+            for context, standard_cols in STANDARD_TABLE_COLUMNS.items():
+                for valid_col in standard_cols:
+                    if valid_col in columns:
+                        index.append(valid_col)
+                if index:
+                    logger.info("Context is %s ", context)
+            logger.debug("Proudly presenting the index: %s", index)
+        else:
+            index = preset_index
+        if "REAL" in columns:
+            index.append("REAL")
+
+        return index
+
+    def _check_index(self, index):
+        """Check the table index
+        ARGS:
+            index (list): list of column names
+
```

Comment:
Punctuation ending first line (which should be more precise, then space, then Args: not ARGS: for google docstring:

And please, use typing annotations

def _check_index(self, index: list):
```
"""Check if valid tablindex is present.

     Args:
            index: list of column names
```

Reply:
About the dockstring: Yes, not sure why that all caps snuck in, gone now. When it comes to the dot, if that is what you want, that is fine

Reply:
When it comes to the unwanteds: changed to not_founds, to make it clearer what they are. The idea is to make the user aware that he/she is trying to use column names that are not present in the table

Reply:
Woops: last comment belongs to issue below: look at it there